src/app/zap-templates/templates/app/cluster-enums-src.zapt

```diff
@@ -0,0 +1,39 @@
+{{> header}}
+
+#include <app-common/zap-generated/cluster-enums.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+{{#zcl_clusters}}
+namespace {{asUpperCamelCase name}} {
+{{#zcl_enums}}
+
+{{#if (isWeaklyTypedEnum label)}}
+// Need to convert consumers to using the new enum classes, so we
+// don't just have casts all over.
+#ifdef CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
+{{/if}}
+CHIP_ERROR Check{{asType label}}({{asType label}} val)
```

Comment:
This is fine, but we should be able to output the Check methods for the weakly typed enums even if CHIP_USE_ENUM_CLASS_FOR_IM_ENUM is not defined.  They just need to look a bit different.  That can be a follow-up, though.